test/dynamo/test_export.py

```diff
@@ -1777,6 +1777,21 @@ def fn_with_kwargs(pos0, tuple0, *myargs):
         real_result = fn_with_kwargs(pos0, tuple0, *myargs)
         self.assertTrue(torch._dynamo.utils.same(real_result, dynamo_result))
 
+    @patch.object(torch._dynamo.config, "dynamic_shapes", True)
+    def test_persist_torch_assert(self):
```

Comment:
Actually this passes without any fix, as @jansel mentioned, the ```SymNodeVariable``` is not constant variable, so won't go through constant fold. See the generated graph:
```
 __compiled_fn_0 <eval_with_key>.1 opcode         name     target                                args               kwargs
-------------  -------  ------------------------------------  -----------------  --------
placeholder    x        x                                     ()                 {}
call_method    size     size                                  (x,)               {}
call_function  getitem  <built-in function getitem>           (size, 0)          {}
call_function  eq       <built-in function eq>                (getitem, 4)       {}
call_function  _assert  <function _assert at 0x7fbd37c195e0>  (eq, 'assertion')  {}
call_function  add      <built-in function add>               (x, 1)             {}
output         output   output                                ((add,),)          {}
```